Sources/Amplitude/Amplitude.swift

```diff
@@ -1,7 +1,7 @@
 import Foundation
 
 public class Amplitude {
-    var configuration: Configuration
+    public var configuration: Configuration
```

Comment:
Previously configurations cannot be accessible after init. 
But notice that the configurations in Kotlin are accessible and be able to be dynamically set.
Check [here](https://github.com/amplitude/Amplitude-Kotlin/pull/132/files) for Kotlin.
Let me know if there have any concerns.

Reply:
Shouldn't `public private(set) var configuration: Configuration` provide external access?

Reply:
If not, I would probably prefer we make our own public accessor and keep the internal var private.

Reply:
Ha, I just tried to use `public private(set)` like [here](https://forums.swift.org/t/can-you-have-public-get-private-set-not-internal-get/40900/2) mentioned. 
But I can still override the configuration value.

Reply:
Talked with Justin offline. 
We aim to prevent the Configuration object from being overridden. This could potentially avoid issues where multiple plugins override the configuration object and point to different configurations object.
Furthermore, we want the customer to be able to overwrite the value in the configuration, which is also in alignment with Kotlin.